Transition Mdbook to Vocs

AgeManning opened the pull request:
I've shifted our docs from mdbook to vocs. You can see a live version here: https://anchor.sigmaprime.io

There a few benefits in doing this:
- It looks a bit nicer, it allows custom CSS and some extra markdown features
- It gives a landing page integrated with the docs
- Is a bit more modern than md-book, so might be easier to maintain

### Important Note:
I am not a designer or front-end dev by any stretch of the imagination. The resulting design here comes from a few iterations of me and my friend.  

I'm happy to make some small changes here and there to the design, but if people don't like the design, I am in no way attached to it at all and more than happy for someone to re-design it. The goal of this PR was to do most of the heavy lifting to a new framework, handle CI and automated upgrades to a production website. Modifying the design from here shouldn't be too hard, have at it :). 

This PR also adds some documentation, but it shouldn't be taken as the final form of documentation (which I think needs some love), we should improve documentation as a future PR. 


Final Note: I've not tested the CI modifications, it will be an ongoing process :)

AgeManning committed:
Initial template, design and most of the upgrade

AgeManning committed:
Improve docs and adjust CI

AgeManning committed:
Modify readme to point to new docs

AgeManning committed:
Remove dist, set the base url correctly

AgeManning committed:
Remove old book workflow

AgeManning committed:
Handle playwright for mermaid diagrams in the CI

AgeManning committed:
Sync github stars and add it to the CI build script.

AgeManning committed:
Add extra terms to spellcheck wordlist

AgeManning committed:
Merge latest unstable

AgeManning committed:
Improve build script to auto-sync contributors

jking-aus commented:
age this looks so good

realbigsean commented:
Vocs is looking great.

Ran into some 404s when I was clicking around, the three sublinks from here: https://anchor.sigmaprime.io/cli
<img width="288" height="143" alt="Screenshot 2025-08-12 at 6 09 34 AM" src="https://github.com/user-attachments/assets/61ccd984-486a-4493-8bb2-99d001ecca34" />

dknopik commented:
Also got one, the "Code of conduct" in the footer:
<img width="468" height="81" alt="image" src="https://github.com/user-attachments/assets/f507396b-e3aa-4124-9a4c-bbe0d6a4338b" />
https://anchor.sigmaprime.io/docs/conduct

AgeManning committed:
Fix routing for multiple domains and contribution link

AgeManning commented:
Ok. 

So, I modified github secrets. When any PR is merged, it updates the current website. So PR's being merged were over-writing the new site. The new site is covered by cloudflare which still servers you the normal html. So most of the bad links were because the site had been partially updated by a PR merge. 

@dknopik's one was a real one tho. I fixed that. I've also re-routed anchor-book.sigmaprime.io, to the current website. So hopefully all is fixed now.

AgeManning committed:
Merge latest unstable

AgeManning committed:
Fix tip in intro

AgeManning committed:
Remove old artifact
